packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -161,13 +162,15 @@ export async function doDeployArmTemplates(ctx: SolutionContext): Promise<Result
     throw new Error("Failed to get resource group from project solution settings.");
   }
 
+  const bicepCommand = await ensureBicep(ctx);
```

Comment:
Maybe move the bicep check step into the `compileBiceoToJson()` method so the method can self-comtain?

Reply:
it depends `ctx`, and `compileBiceoToJson` don't need `ctx`.